src/codegen/doc/mod.rs

```diff
@@ -517,13 +561,34 @@ where
             }
         }
 
-        if let Some(ref doc) = fn_.ret().doc {
+        let out_parameters: Vec<_> = no_array_length_params
+            .iter()
+            .filter(|param| {
+                param.direction == ParameterDirection::Out
+                    && !(param.name == "error" && param.c_type == "GError**")
+            })
+            .collect();
+
+        if fn_.ret().doc.is_some() || !out_parameters.is_empty() {
             writeln!(w, "\n# Returns\n")?;
-            writeln!(
-                w,
-                "{}",
-                reformat_doc(&fix_param_names(doc, &self_name), &symbols, &parent_name)
-            )?;
+            if let Some(ref doc) = fn_.ret().doc {
+                writeln!(
+                    w,
+                    "{}",
+                    reformat_doc(&fix_param_names(doc, &self_name), &symbols, &parent_name)
+                )?;
+            }
+            // fold out parameters as part of returns
+            for parameter in out_parameters {
```

Comment:
Should there be a newline between return doc and return parameters just like there are two empty lines around `# Returns`? Doesn't make a difference for the rendered docs, but makes the markdown source easier to read.

Reply:
Why there would be? they are part of the returns

Reply:
Nicer to read. Currently:

```rust
    /// # Returns
    ///
    /// TRUE if the conversion was successful.
    /// ## `dest_val`
    /// pointer to destination value
```
And I'd like:

```rust
    /// # Returns
    ///
    /// TRUE if the conversion was successful.
    ///
    /// ## `dest_val`
    /// pointer to destination value
```
But again, this makes no difference to **rendered markdown output**, which is all we care about. Feel free to ignore this and mark it as resolved.

Reply:
Heck, we might even remove the `\n` from `# Returns` to save quite a few characters/lines in this intermediate textual form.

Reply:
Ah I see what you mean. Yeah, that would make the docs output readable